File: py-scoring/src/ai/h2o/sparkling/ml/params/HasFeatureTypesOnMOJO.py
from ai.h2o.sparkling.ml.params.H2OTypeConverters import H2OTypeConverters


class HasFeatureTypesOnMOJO:

    def getFeatureTypes(self):
        return H2OTypeConverters.scalaMapStringStringToDictStringAny(self._java_obj.getFeatureTypes())

File: scoring/src/main/scala/ai/h2o/sparkling/ml/params/HasFeatureTypesOnMOJO.scala
package ai.h2o.sparkling.ml.params

import org.apache.spark.ml.param.Params

trait HasFeatureTypesOnMOJO extends Params {
  protected final val featureTypes: MapStringStringParam =
    new MapStringStringParam(this, "featureTypes", "Types of feature columns expected by the model")

  def getFeatureTypes(): Map[String, String] = $(featureTypes)

  setDefault(featureTypes -> Map.empty[String, String])
}
